be/src/exec/query_cache/ticket_checker.h

```diff
@@ -0,0 +1,56 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+#pragma once
+
+#include <glog/logging.h>
+
+#include <memory>
+#include <unordered_map>
+
+#include "gutil/macros.h"
+namespace starrocks::query_cache {
+
+// TicketChecker is used to count down EOS chunk of the SplitMorsels generated from the same
+// original ScanMorsel.
+// when we split a SplitMorsel from the ScanMorsel, we invoke TicketChecker::enter to increment
+// enter_count by 1. if the SplitMorsel is the last one, we should turn on all_ready_bit of
+// the TicketChecker. when ScanOperator receives an EOS chunk, we invoke TicketChecker::leave to
+// increment leave_count by 1 then check if the all_ready_bit is on and enter_count is equivalent
+// to the leave_count, if that is true, we know ScanOperator receives all of the EOS chunks of
+// SplitMorsels, that means the original ScanMorsel has been processed completely and a EOS chunk
+// can emit from ScanOperator to its successor MultilaneOperator.
+
+class TicketChecker;
+using TicketCheckerRawPtr = TicketChecker*;
+using TicketCheckerPtr = std::shared_ptr<TicketChecker>;
+using TicketIdType = int64_t;
+
+class TicketChecker {
```

Comment:
such as MorselEos name is better?

Reply:
TicketChecker is inspired by team admission tickets,  when all of the team members gather together, the show there team ticket, the ticket checker permit them enter, after they leaves, show the team ticket again ant the ticket is collected and invalid.